lib/TPP/Transforms/TileConfig.cpp

```diff
@@ -0,0 +1,154 @@
+#include "TPP/Dialect/Xsmm/XsmmUtils.h"
+#include "TPP/Transforms/Utils/VNNIUtils.h"
+#include "mlir/Dialect/Func/IR/FuncOps.h"
+#include "mlir/Dialect/MemRef/IR/MemRef.h"
+#include "mlir/Dialect/SCF/IR/SCF.h"
+#include "mlir/Dialect/SCF/Transforms/Transforms.h"
+#include "mlir/Pass/Pass.h"
+#include "mlir/Transforms/GreedyPatternRewriteDriver.h"
+#include <iostream>
+#include <list>
+namespace mlir {
+namespace tpp {
+#define GEN_PASS_DEF_TILECONFIGINSERTIONPASS
+#include "TPP/Passes.h.inc"
+} // namespace tpp
+} // namespace mlir
+
+using namespace mlir;
+
+namespace mlir {
+namespace tpp {
+
+template <typename InvokeOpTy, typename DispatchOpTy>
+static void appendBrgemmFlags(SmallVector<Attribute> &attributes,
+                              PatternRewriter &rewriter, InvokeOpTy opTy) {
+  auto flags =
+      dyn_cast<DispatchOpTy>(opTy.getOperand(0).getDefiningOp()).getFlags();
+  for (auto flagItr : flags) {
+    if (flagItr == xsmm::GemmFlagsAttr::get(rewriter.getContext(),
+                                            xsmm::GemmFlags::NONE)) {
+      return;
+    }
+    attributes.push_back(flagItr);
+  }
+
+  if (attributes.empty())
+    attributes.push_back(
+        xsmm::GemmFlagsAttr::get(rewriter.getContext(), xsmm::GemmFlags::NONE));
+}
+
+template <typename InvokeOpTy, typename DispatchOpTy>
+struct TileConfig : OpRewritePattern<InvokeOpTy> {
+  using OpRewritePattern<InvokeOpTy>::OpRewritePattern;
+
+  LogicalResult matchAndRewrite(InvokeOpTy op,
+                                PatternRewriter &rewriter) const override {
+    if (xsmm::utils::getDataType(rewriter, op.getOperand(1).getType()) !=
+        xsmm::DataTypeAttr::get(rewriter.getContext(), xsmm::DataType::BF16))
+      return failure();
+    auto flags =
+        dyn_cast<DispatchOpTy>(op.getOperand(0).getDefiningOp()).getFlags();
+    for (auto flagItr : flags) {
+      if (flagItr == xsmm::GemmFlagsAttr::get(
+                         rewriter.getContext(),
+                         mlir::xsmm::GemmFlags::NO_RESET_TILECONFIG) ||
+          flagItr == xsmm::GemmFlagsAttr::get(
+                         rewriter.getContext(),
+                         mlir::xsmm::GemmFlags::NO_SETUP_TILECONFIG)) {
+        return failure();
+      }
+    }
+
+    SmallVector<Attribute> attributesSetup;
+    attributesSetup.push_back(xsmm::GemmFlagsAttr::get(
+        rewriter.getContext(), xsmm::GemmFlags::NO_RESET_TILECONFIG));
+    appendBrgemmFlags<InvokeOpTy, DispatchOpTy>(attributesSetup, rewriter, op);
+    auto tileConfigSetup = rewriter.create<xsmm::TileConfigDispatchOp>(
+        op.getLoc(), rewriter.getI64Type(),
+        DenseI64ArrayAttr::get(
+            rewriter.getContext(),
+            dyn_cast<DispatchOpTy>(op.getOperand(0).getDefiningOp())
+                .getInputs()),
+        rewriter.getArrayAttr(attributesSetup),
+        xsmm::utils::getDataType(rewriter, op.getOperand(1).getType()));
+
+    SmallVector<Attribute> attributesReset;
```

Comment:
Why are you adding both here, all to the same `op`?

Reply:
The NO_RESET_TILECONFIG flag is added, followed by the rest of the flags of the `op` appended via the `appendBrgemmFlags` call.